src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -2110,15 +2132,16 @@ private void UpperBoundInference(TypeSymbol source, TypeSymbol target, ref HashS
             // SPEC: * Otherwise, if V is nullable type V1? and U is nullable type U1?
             // SPEC:   then an exact inference is made from U1 to V1.
 
-            if (ExactNullableInference(source, target, ref useSiteDiagnostics))
-            {
-                return;
-            }
+            Debug.Assert(source.IsReferenceType);
+            //if (ExactNullableInference(source, target, ref useSiteDiagnostics))
+            //{
+            //    return;
+            //}
```

Comment:
I want it to stay. This code is subtle and we only do not need to do the commented checks because we do not get value types here. And thus the assert above.

I will add a comment that spec asks us to do the following, but the current implementation handles all value types in the callers, so we do not have to.